sound/soc/sof/trace.c

```diff
@@ -566,23 +573,42 @@ void snd_sof_release_trace(struct snd_sof_dev *sdev)
 			dev_err(sdev->dev, "DMA_TRACE_FREE failed with error: %d\n", ret);
 	}
 
+	if (only_stop)
+		goto out;
+
 	ret = snd_sof_dma_trace_release(sdev);
 	if (ret < 0)
 		dev_err(sdev->dev,
 			"error: fail in snd_sof_dma_trace_release %d\n", ret);
 
-	sdev->dtrace_is_enabled = false;
+	sdev->dtrace_state = SOF_DTRACE_DISABLED;
+
+out:
 	sdev->dtrace_draining = true;
 	wake_up(&sdev->trace_sleep);
 }
-EXPORT_SYMBOL(snd_sof_release_trace);
+
+int snd_sof_trace_suspend(struct snd_sof_dev *sdev, pm_message_t pm_state)
```

Comment:
Maybe make this return void too?

Reply:
@ranj063 Not sure which is good, void or int? I saw ```sof_suspend_clients()``` returns int. To make them aligned, I used int. @ujfalusi What's your option?

Reply:
Yes, the suspend for the clients also returns `int` but it ignores the return value from the auxdev's suspend.
I'm not sure what would we do if something fail. In most cases this is problematic if we keep the DSP on.
Not sure, I think the clients should be returning error and if it make sense for the dtrace, it should also, if not then make it `void`